server/checks/Specification.js

```diff
@@ -132,27 +136,52 @@ export default class Specification extends Check {
       [MINIMUM_LENGTH]: {
         enabled: shouldCheckLength,
         fn: isLongEnough.bind(null, body, requiredLength)
+      },
+      [DIFFERS_FROM_TEMPLATE]: {
+        enabled: shouldCheckPrTemplate,
+        fn: this._differsFromPrTemplate.bind(this, body, user, repo, token)
       }
     }
 
     // array to force the order
-    const [success, failedChecks] = [CONTAINS_ISSUE_NUMBER, CONTAINS_URL, MINIMUM_LENGTH]
-      .reduce(([success, failedChecks], checkName) => {
-        const { enabled, fn: check } = checksMapping[checkName]
-        if (enabled) {
-          const res = check()
-          if (!res) {
-            failedChecks.push(`'${checkName}'`)
-          }
-
-          success = success || res
+    const [success, failedChecks] = await [
+      CONTAINS_ISSUE_NUMBER, CONTAINS_URL, DIFFERS_FROM_TEMPLATE, MINIMUM_LENGTH
+    ].reduce(async(awaitable, checkName) => {
+      const {enabled, fn: check} = checksMapping[checkName]
+      let [success, failedChecks] = await awaitable
+
+      if (enabled) {
+        const res = await check()
+        if (!res) {
+          failedChecks.push(`'${checkName}'`)
         }
 
-        return [success, failedChecks]
+        success = success || res
+      }
+
+      return [success, failedChecks]
     }, [undefined, []])
 
     if (!success) {
       throw new Error(`PR's body failed check ${failedChecks[0]}`)
     }
   }
+
+  /**
+   * @param {string} content to compare
+   * @param {string} user
+   * @param {string} repo
+   * @param {string} accessToken
+   *
+   * @return {boolean} true if `content` is not equal to template. False otherwise
+   *
+   * @private
+   */
+  _differsFromPrTemplate(content, user, repo, accessToken) {
+    return this.github.readPullRequestTemplate(user, repo, accessToken)
+      .then((template) => template !== content, () => {
+        info(`${user}/${repo}: No PULL_REQUEST_TEMPLATE found`)
+        return false
```

Comment:
I guess you could argue that, yes :) However we would enable this check by default in our Zappr Enterprise config, without everybody actually using PR templates. Imo it should just mean "in case you're using a template, the PR body must be different."